### PR TITLE
set properties about and MiddleName to nullable in validator of editU…

### PR DESCRIPTION
**Pull request by Cheephot**
…ser request
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/892

### CODE REVIEW

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 1]
```diff
@@ -141,7 +141,6 @@ private void HandleInternalPropertyValidation(Operation<EditUserRequest> request
                 new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
```
Comment (MariyaSin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:141]: rework use BaseEditRequestValidator from kernel

Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs:101]: done

**Review by BelehovEgor — APPROVED** [review]

**Review by VLiamin — APPROVED** [review]

**Review by borovHokage — APPROVED** [review]

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 2]
```diff
@@ -4,64 +4,24 @@
...
-                        context.AddFailure(propertyName, validateDelegate.Value);
-                    }
-                }
-            }
-
-            #endregion
+            Context = context;
+            RequestedOperation = requestedOperation;
 
             #region paths
```
Comment (MariyaSin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:27]: fix op of MiddleName, City, About

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 3]
```diff
@@ -141,7 +101,6 @@ private void HandleInternalPropertyValidation(Operation<EditUserRequest> request
                 new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
```
Comment (MariyaSin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:101]: remove x == OperationType.Add

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 2 above)
Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs:27]: done

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 3 above)
Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs:101]: fix

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 4]
```diff
@@ -1,376 +1,309 @@
...
 
-        private void HandleInternalPropertyValidation(Operation<EditUserRequest> requestedOperation, CustomContext context)
+      #region firstname
+
+      AddFailureForPropertyIf(
+        nameof(EditUserRequest.FirstName),
+        x => x == OperationType.Replace,
+        new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
+        {
+          { x => !NameRegex.IsMatch(x.value.ToString()), "First name must not contain numbers" },
```
Comment (MariyaSin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:75]: nulRef if null

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 5]
```diff
@@ -1,376 +1,309 @@
...
+          { x => !string.IsNullOrEmpty(x.value.ToString()), "First name is empty" },
```
Comment (MariyaSin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:76]: "First name must not be empty."

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 6]
```diff
@@ -1,376 +1,309 @@
...
+          { x => x.value.ToString().Length < 32, "First name is too long" }
```
Comment (MariyaSin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:77]: nullRf if null

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 7]
```diff
@@ -1,376 +1,309 @@
...
+      #region lastname
+
+      AddFailureForPropertyIf(
+        nameof(EditUserRequest.LastName),
+        x => x == OperationType.Replace,
+        new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
         {
-            #region local functions
+          { x => !NameRegex.IsMatch(x.value.ToString()), "Last name must not contain numbers" },
+          { x => !string.IsNullOrEmpty(x.value.ToString()), "Last name is empty" },
```
Comment (MariyaSin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:90]: "Last name must not be empty."

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 8]
```diff
@@ -1,376 +1,309 @@
...
+
+      #region lastname
+
+      AddFailureForPropertyIf(
+        nameof(EditUserRequest.LastName),
+        x => x == OperationType.Replace,
+        new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
         {
-            #region local functions
+          { x => !NameRegex.IsMatch(x.value.ToString()), "Last name must not contain numbers" },
```
Comment (MariyaSin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:89]: nullRef if null

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 9]
```diff
@@ -1,376 +1,309 @@
...
+          { x => !string.IsNullOrEmpty(x.value.ToString()), "Last name is empty" },
+          { x => x.value.ToString().Length < 100, "Last name is too long" }
```
Comment (MariyaSin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:91]: nullRef if null

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 4 above)
Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs]: done

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 6 above)
Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs]: done

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 7 above)
Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs]: done

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 8 above)
Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs]: done

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 9 above)
Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs]: done

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 5 above)
Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs]: done

**Pull request closed without merging**